pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,264 @@
+	AccountRole    string `hcl:"org_account_role"`
+	AccountRegion  string `hcl:"org_account_region"`
+	AccountListTTL string `hcl:"org_account_map_ttl"`
+}
+
+type organizationValidation struct {
+	orgListAccountMap             map[string]bool
+	orgListAccountMapCreationTime time.Time
+	orgConfig                     *orgValidationConfig
+	mutex                         sync.RWMutex
+	orgAccountListCacheTTL        time.Duration // set from configuration else will be set to default
+	log                           hclog.Logger
+	retries                       int // fix number of retries before ttl is expired.
+}
+
+func newOrganizationValidationBase(config *orgValidationConfig) *organizationValidation {
+	client := &organizationValidation{
+		orgListAccountMap: make(map[string]bool),
+		orgConfig:         config,
+		retries:           orgAccountRetries,
+	}
+
+	return client
+}
+
+func (o *organizationValidation) getRetries() int {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+	return o.retries
+}
+
+func (o *organizationValidation) decrRetries() int {
+	// Only decr if its greater than 0
+	if o.retries > 0 {
+		o.retries -= 1
+	}
+
+	return o.retries
+}
+
+func (o *organizationValidation) configure(config *orgValidationConfig) error {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.orgConfig = config
+
+	// While doing configuration invalidate the map so we dont keep using old one.
+	o.orgListAccountMap = make(map[string]bool)
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.Internal, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
+	}
+
+	o.orgAccountListCacheTTL = t
+
+	return nil
+}
+
+func (o *organizationValidation) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// This verification method checks if the Account ID attached on the node is part of the organisation.
+// If it part of the organisation then validation should be succesfull if not attestation should fail, on enabling this verification method.
+// This could be alternative for not explictly maintaing allowed list of account ids.
+// Method pulls the list of accounts from organization and caches it for certain time, cache time can be configured.
+func (o *organizationValidation) ValidateAccountBelongstoOrg(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string) (bool, error) {
+
+	orgAccountList, isStale, err := o.checkIfOrgAccountListIsStale(ctx)
+	if err != nil {
+		return false, err
+	}
+	// refresh the account map
+	if isStale {
+		orgAccountList, err = o.fetchAccountsListFromOrg(ctx, orgClient, false)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	// Check if account contains in organization list.
+	// If this return false we will check `X` number of times in that window, to avoid
+	// genuine accounts which are newly added
+	exist := checkIfAccountIdExist(orgAccountList, accoundIDofNode)
+	if !exist {
```

Comment:
There should probably be dedicated functions for these kinds of things, like `o.reloadAccountList` and `o.cacheMiss()`, where the latter can conditionally call the former.

Reply:
Got it, added new function to handle this.